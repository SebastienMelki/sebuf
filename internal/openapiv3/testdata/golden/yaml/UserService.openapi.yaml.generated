openapi: 3.1.0
info:
    title: UserService API
    version: 1.0.0
paths:
    /api/v1/users/create:
        post:
            tags:
                - UserService
            summary: CreateUser
            description: Create user
            operationId: CreateUser
            parameters:
                - name: X-User-Token
                  in: header
                  description: User authentication token
                  required: true
                  schema:
                    type: string
                    format: uuid
            requestBody:
                content:
                    application/json:
                        schema:
                            $ref: '#/components/schemas/User'
                required: true
            responses:
                "200":
                    description: Successful response
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/User'
                "400":
                    description: Validation error
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/ValidationError'
                default:
                    description: Error response
                    content:
                        application/json:
                            schema:
                                type: object
                                properties:
                                    error:
                                        type: string
                                    code:
                                        type: integer
    /api/v1/users/get:
        post:
            tags:
                - UserService
            summary: GetUser
            description: Get user
            operationId: GetUser
            parameters:
                - name: X-User-Token
                  in: header
                  description: User authentication token
                  required: true
                  schema:
                    type: string
                    format: uuid
            requestBody:
                content:
                    application/json:
                        schema:
                            $ref: '#/components/schemas/Request'
                required: true
            responses:
                "200":
                    description: Successful response
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/User'
                "400":
                    description: Validation error
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/ValidationError'
                default:
                    description: Error response
                    content:
                        application/json:
                            schema:
                                type: object
                                properties:
                                    error:
                                        type: string
                                    code:
                                        type: integer
    /api/v1/users/update:
        post:
            tags:
                - UserService
            summary: UpdateUser
            description: Update user
            operationId: UpdateUser
            parameters:
                - name: X-User-Token
                  in: header
                  description: User authentication token
                  required: true
                  schema:
                    type: string
                    format: uuid
            requestBody:
                content:
                    application/json:
                        schema:
                            $ref: '#/components/schemas/User'
                required: true
            responses:
                "200":
                    description: Successful response
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/User'
                "400":
                    description: Validation error
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/ValidationError'
                default:
                    description: Error response
                    content:
                        application/json:
                            schema:
                                type: object
                                properties:
                                    error:
                                        type: string
                                    code:
                                        type: integer
components:
    schemas:
        FieldViolation:
            type: object
            properties:
                field:
                    type: string
                    description: The field path that failed validation (e.g., 'user.email' for nested fields). For header validation, this will be the header name (e.g., 'X-API-Key')
                description:
                    type: string
                    description: Human-readable description of the validation violation (e.g., 'must be a valid email address', 'required field missing')
            required:
                - field
                - description
            description: FieldViolation describes a single validation error for a specific field.
        ValidationError:
            type: object
            properties:
                violations:
                    type: array
                    items:
                        $ref: '#/components/schemas/FieldViolation'
                    description: List of validation violations
            required:
                - violations
            description: ValidationError is returned when request validation fails. It contains a list of field violations describing what went wrong.
        User:
            type: object
            properties:
                id:
                    type: string
                    description: User ID
                name:
                    type: string
                    description: User name
                email:
                    type: string
                    description: User email
                role:
                    type: string
                    description: User role
            description: User message used by multiple services
        Request:
            type: object
            properties:
                id:
                    type: string
                    description: Request ID
                data:
                    type: string
                    description: Request data
            description: Generic request message
