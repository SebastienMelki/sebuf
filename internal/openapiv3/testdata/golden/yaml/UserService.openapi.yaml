openapi: 3.1.0
info:
    title: UserService API
    version: 1.0.0
paths:
    /api/v1/users/create:
        post:
            tags:
                - UserService
            summary: CreateUser
            description: Create user
            operationId: CreateUser
            parameters:
                - name: X-User-Token
                  in: header
                  description: User authentication token
                  required: true
                  schema:
                    type: string
                    format: uuid
            requestBody:
                content:
                    application/json:
                        schema:
                            $ref: '#/components/schemas/User'
                required: true
            responses:
                "200":
                    description: Successful response
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/User'
                default:
                    description: Error response
                    content:
                        application/json:
                            schema:
                                type: object
                                properties:
                                    error:
                                        type: string
                                    code:
                                        type: integer
    /api/v1/users/get:
        post:
            tags:
                - UserService
            summary: GetUser
            description: Get user
            operationId: GetUser
            parameters:
                - name: X-User-Token
                  in: header
                  description: User authentication token
                  required: true
                  schema:
                    type: string
                    format: uuid
            requestBody:
                content:
                    application/json:
                        schema:
                            $ref: '#/components/schemas/Request'
                required: true
            responses:
                "200":
                    description: Successful response
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/User'
                default:
                    description: Error response
                    content:
                        application/json:
                            schema:
                                type: object
                                properties:
                                    error:
                                        type: string
                                    code:
                                        type: integer
    /api/v1/users/update:
        post:
            tags:
                - UserService
            summary: UpdateUser
            description: Update user
            operationId: UpdateUser
            parameters:
                - name: X-User-Token
                  in: header
                  description: User authentication token
                  required: true
                  schema:
                    type: string
                    format: uuid
            requestBody:
                content:
                    application/json:
                        schema:
                            $ref: '#/components/schemas/User'
                required: true
            responses:
                "200":
                    description: Successful response
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/User'
                default:
                    description: Error response
                    content:
                        application/json:
                            schema:
                                type: object
                                properties:
                                    error:
                                        type: string
                                    code:
                                        type: integer
components:
    schemas:
        User:
            type: object
            properties:
                id:
                    type: string
                    description: User ID
                name:
                    type: string
                    description: User name
                email:
                    type: string
                    description: User email
                role:
                    type: string
                    description: User role
            description: User message used by multiple services
        Request:
            type: object
            properties:
                id:
                    type: string
                    description: Request ID
                data:
                    type: string
                    description: Request data
            description: Generic request message
        Response:
            type: object
            properties:
                success:
                    type: boolean
                    description: Success indicator
                message:
                    type: string
                    description: Response message
                data:
                    type: string
                    description: Response data
            description: Generic response message
